domain/application/service/storage/directive.go

```diff
@@ -60,6 +60,19 @@ func (s *Service) GetApplicationStorageDirectives(
 	return s.st.GetApplicationStorageDirectives(ctx, uuid)
 }
 
+// GetApplicationStorage returns the storage directives set for an application,
+// keyed to the storage name. If the application does not have any storage
+// directives set then an empty result is returned.
+//
+// If the application does not exist, then a [applicationerrors.ApplicationNotFound]
+// error is returned.
+func (s *Service) GetApplicationStorage(
+	ctx context.Context,
+	uuid coreapplication.UUID,
+) (application.ApplicationStorage, error) {
```

Comment:
Can you validate the uuid `uuid.Validate` and return a `coreerrors.NotValid` if it fails.

Reply:
Yup, will update `GetApplicationStorageDirectives` too.